account_payment_order/models/account_invoice.py

```diff
@@ -0,0 +1,111 @@
+# © 2013-2014 ACSONE SA (<https://acsone.eu>).
+# © 2014 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    payment_order_ok = fields.Boolean(
+        related='payment_mode_id.payment_order_ok', readonly=True)
+
+    @api.model
+    def _get_reference_type(self):
+        rt = super(AccountInvoice, self)._get_reference_type()
+        rt.append(('structured', _('Structured Reference')))
+        return rt
+
+    @api.model
+    def line_get_convert(self, line, part):
+        """Copy supplier bank account from invoice to account move line"""
+        res = super(AccountInvoice, self).line_get_convert(line, part)
+        if line.get('type') == 'dest' and line.get('invoice_id'):
+            invoice = self.browse(line['invoice_id'])
+            if invoice.type in ('in_invoice', 'in_refund'):
+                res['partner_bank_id'] = invoice.partner_bank_id.id or False
+        return res
+
+    @api.multi
+    def _prepare_new_payment_order(self):
+        self.ensure_one()
+        vals = {'payment_mode_id': self.payment_mode_id.id}
+        # other important fields are set by the inherit of create
+        # in account_payment_order.py
+        return vals
+
+    @api.multi
+    def create_account_payment_line(self):
+        apoo = self.env['account.payment.order']
+        aplo = self.env['account.payment.line']
+        result_payorder_ids = []
+        action_payment_type = 'debit'
+        for inv in self:
+            if inv.state != 'open':
+                raise UserError(_(
+                    "The invoice %s is not in Open state") % inv.number)
+            if not inv.payment_mode_id:
+                raise UserError(_(
+                    "No Payment Mode on invoice %s") % inv.number)
+            if not inv.move_id:
+                raise UserError(_(
+                    "No Journal Entry on invoice %s") % inv.number)
+            if not inv.payment_order_ok:
+                raise UserError(_(
+                    "The invoice %s has a payment mode '%s' "
```

Comment:
no string formatting:
"which is not selectable in payment orders.") % (inv.number, inv.payment_mode_id.display_name))